### PR TITLE
Reflection_oM: QuantityAttribute expanded to InputClassificationAttribute, FolderPathAttribute and FilePathAttribute added

### LINKED ISSUE #1067: Quantity_oM, Reflection_oM: expand QuantityAttribute to InputClassificationAttribute
#### Description:

As discussed with @al-fisher, @adecler and @IsakNaslundBh, it would be good to enable storing more general information about the input characteristics, not only their quantities. Therefore, `Quantity` property of input- and output-related attributes could be changed to `Classification` property of type `InputClassificationAttribute`, from which `QuantityAttribute` will inherit.

This will allow adding other attributes inheriting from `InputClassificationAttribute`, such as `FolderPathAttribute` or `FilePathAttribute`.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #1067

### Test files

See https://github.com/BHoM/BHoM_Engine/pull/2132.

### Changelog

- `InputClassificationAttribute` added
- `QuantityAttribute` made inherit from `InputClassificationAttribute`
- `FolderPathAttribute` and `FilePathAttribute` added

### Additional comments

A blessing has been given by @FraserGreenroyd to open this PR and merge in parallel with #1069. No other potential conflicts detected.

### CODE REVIEW

**Review by michaelhoehn — DISMISSED** [review]
Tested functionality on a few random objects with success. LGTM!

File: Reflection_oM/Attributes/IInputClassificationAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using BH.oM.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace BH.oM.Reflection.Attributes
+{
+    public abstract class InputClassificationAttribute : Attribute
```
Comment (FraserGreenroyd) [inline Reflection_oM/Attributes/IInputClassificationAttribute.cs:9]: Class name is not an interface, but file name is `IInputClassificationAttribute` with a leading I. One or the other needs fixing please :smile:

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check code compliance

**Review by FraserGreenroyd — APPROVED** [review]
Am happy now, thanks! :smile:

**Comment by pawelbaran** [comment]
@adecler is there a chance you could review the chain of PRs starting here (this one, https://github.com/BHoM/BHoM_Engine/pull/2132, https://github.com/BHoM/BHoM_UI/pull/333)? Would be good to get your blessing before merge. Thanks!

**Review by adecler — APPROVED** [review]
Looks good to me

**Comment by FraserGreenroyd** [comment]
I expect this to fail but @BHoMBot check installer please

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Pull request merged**